source/agora/consensus/state/UTXOSet.d

```diff
@@ -190,6 +235,130 @@ public class UTXOSet
         log.trace("findUTXO: utxo_hash {} not found", utxo);
         return false;
     }
+
+    /// Ditto
+    public override UTXOFinder getUTXOFinder () nothrow @trusted
+    {
+        this.used_utxos.clear();
+        return &this.findUTXO;
+    }
+
+    /// Ditto
+    protected override bool peekUTXO (Hash utxo, out UTXO value) nothrow @safe
+    {
+        return this.utxo_db.find(utxo, value);
+    }
+
+    /// Ditto
+    protected override void remove (Hash utxo) @safe
+    {
+        this.utxo_db.remove(utxo);
+    }
+
+    /// Ditto
+    protected override void add (Hash utxo, UTXO value) @safe
+    {
+        this.utxo_db[utxo] = value;
+    }
+}
+
+/*******************************************************************************
+
+    This is a simple UTXOSet, used when the AA behavior is desired
+
+    Most unittestsdo not need a full-fledged UTXOSet with all the DB and
```

Comment:
```suggestion
    Most unittests do not need a full-fledged `UTXOSet` with all the DB and
```

Also, `fully-fledged` ? CC @hewison-chris .

Reply:
British is fully fledged :-)